### PR TITLE
A global way to get token info in gateway

### PR DESCRIPTION
**Pull request by boyuan-chen**
:clipboard: Add associated issues, tickets, docs URL here. https://github.com/bobanetwork/boba/issues/404

## Overview

Describe what your Pull Request is about in a few sentences.

Some tokens are not ERC20 compatible, which means that our standard ABI doesn't work for those tokens.

In order to solve the problem, we store token information in one separate file. In our gateway, it can load token information instead of sending PRC calls. This should improve the speed too.

## Changes

Describe your changes and implementation choices. More details make PRs easier to review.

- Add token information files
- Load name, symbol and decimal from the information file

## Testing

Describe how to test your new feature/bug fix and if possible, a step by step guide on how to demo this.

### CODE REVIEW

File: packages/boba/gateway/src/services/networkService.js [hunk 1]
```diff
@@ -142,7 +146,8 @@ if (APP_CHAIN === 'rinkeby') {
     L1LPAddress: addresses_Mainnet.Proxy__L1LiquidityPool,
     L2LPAddress: addresses_Mainnet.Proxy__L2LiquidityPool,
     ...bobaBridges,
-    ...l0Protocols
+    ...l0Protocols,
+    tokenInfo: tokenInfoMainnet,
```
Comment (sk-enya) [inline packages/boba/gateway/src/services/networkService.js]: Instead of putting this in allAddress we should use separate var for tokenInfo.

**Review by sk-enya — APPROVED** [review]

File: packages/boba/register/addresses/tokenInfoMainnet.json [hunk 2]
```diff
@@ -0,0 +1,276 @@
+{
```
Comment (InoMurko) [inline packages/boba/register/addresses/tokenInfoMainnet.json]: if you intend to use this on alt-l1s - perhaps it makes sense to name the files by chainid?
<chainid>-mainnet.json
<chainid>-testnet.json

Comment (boyuan-chen) [reply packages/boba/register/addresses/tokenInfoMainnet.json]: For alt-l1s, I will use `tokenInfo[L1_NAME][MAINNET or TESTNET]`. It's hard to remember 10 chain Ids.

File: packages/boba/gateway/src/services/networkService.js (hunk 1 above)
Comment (boyuan-chen) [reply packages/boba/gateway/src/services/networkService.js]: Fixed!

File: packages/boba/register/addresses/tokenInfoMainnet.json (hunk 2 above)
Comment (InoMurko) [reply packages/boba/register/addresses/tokenInfoMainnet.json]: well... you can make a mapping of chains, chainids as well. Better this way than having different implementations across the stack

Comment (InoMurko) [reply packages/boba/register/addresses/tokenInfoMainnet.json]: and you get the chainid with rpc and immediately know what to load

Comment (boyuan-chen) [reply packages/boba/register/addresses/tokenInfoMainnet.json]: You mean adding all token information with chain Ids in the `tokenInfoMainnet` file? Like this:
```
{
  chain_id_1: {},
  chain_id_2: {}
}
```

Comment (boyuan-chen) [reply packages/boba/register/addresses/tokenInfoMainnet.json]: > and you get the chainid with rpc and immediately know what to load

The loading process is before connecting to the RPC endpoint. I need to double check if you want to automatically load the file.

Comment (InoMurko) [reply packages/boba/register/addresses/tokenInfoMainnet.json]: yeah that would work too, tokenInfo.json as chainid doesn't tell you if it's testnet or not

Comment (Boyuan-Chen) [reply packages/boba/register/addresses/tokenInfoMainnet.json:1]: One for mainnet and one for testnet. That will be simpler and it should be global for alt l1s except adding token information to the file.

Comment (boyuan-chen) [reply packages/boba/register/addresses/tokenInfoMainnet.json]: Ok. Now, we only need to add token information to `tokenInfo.json` for alt l1s. The gateway automatically fetches the right information based on the chain Id.

**Review by InoMurko — APPROVED** [review]
🙌

**Pull request merged**